File: core/client/game/src/commonGame/scene/base/ILogic.cs
using System;
using ShineEngine;

/// <summary>
/// 逻辑体
/// </summary>
public interface ILogic
{
	/// <summary>
	/// 构造
	/// </summary>
	void construct();

	/// <summary>
	/// 初始化
	/// </summary>
	void init();

	/// <summary>
	/// 析构
	/// </summary>
	void dispose();

	/// <summary>
	/// 刷帧
	/// </summary>
	void onFrame(int delay);
}

File: core/server/project/commonBase/src/main/java/com/home/commonBase/scene/base/ILogic.java
package com.home.commonBase.scene.base;

/** 逻辑体接口 */
public interface ILogic
{
	/** 构造 */
	public void construct();

	/** 初始化(与dispose成对) */
	public void init();

	/** 析构(与dispose成对) */
	public void dispose();

	/** 刷帧 */
	public void onFrame(int delay);
}
